### PR TITLE
locked updates are now not shown

### PR DESCRIPTION
**Pull request by fdellwing**
We have a lot of locked modules here. I excluded them from showing.

Maybe add a switch to do so?

### CODE REVIEW

**Comment by cytopia** [comment]
I have never seen that before. Is this drupal 8 specific?
U might have a screenshot to share that shows the locked modules?

**Comment by fdellwing** [comment]
The output looks like so:

advagg,7.x-2.18,7.x-2.19,Locked via drush. (Aktualisierung verfügbar)
drupal,7.51,7.52,SECURITY UPDATE available
simple_gmap,7.x-1.2,7.x-1.3,Aktualisierung verfügbar

But I got a big derp in my version, just fixxed it and will upload it soon.

**Comment by fdellwing** [comment]
Still not implemented a switch for this. Should I?

**Comment by fdellwing** [comment]
For the locking feature:

I'm not quite sure about this, but it seems to be part in drush since version 6

drush pm-updatestatus -lock=module
drush pm-updatestatus -unlock=module

**Comment by fdellwing** [comment]
Here you go :)

File: bin/check_drupal [hunk 1]
```diff
@@ -548,8 +552,6 @@ print_help() {
 	printf "  -n <name>              [optional] Specify a name for the drupal instance to\n"
 	printf "                         appear on the nagios output. The default is 'Drupal'\n\n"
 
-	printf "                         the 'drupal' folder. This parameter is required.\n\n"
```
Comment (daften) [inline bin/check_drupal:546]: Why is this gone ? :o

Comment (fdellwing) [reply bin/check_drupal:546]: Because it does not belongs there ;)

Run the help and you see it.

Comment (daften) [reply bin/check_drupal:546]: Ah, yes, my apologies, didn't read thoroughly enough :)
My OCD side says this should be a separate PR. My honest side says I'd do it in here too :D

**Comment by fdellwing** [comment]
I'm going to clock off for today, will be back tomorrow 9am.

This should be finished now, but please look through it carefully. I'm not a sh pro ;)

File: bin/check_drupal [hunk 2]
```diff
@@ -293,6 +293,11 @@ check_security_updates() {
 		return $EXIT_UNKNOWN
 	fi
 
+	# Remove via drush locked updates, keep the rest
```
Comment (cytopia) [inline bin/check_drupal:296]: Why is this code block included twice?
Here (Line 296) and also Line 342?

**Comment by cytopia** [comment]
I would also suggest that locked updates should be shown in extended output, so that u at least know that they exist.

File: bin/check_drupal (hunk 2 above)
Comment (fdellwing) [reply bin/check_drupal:296]: Because there are two different functions? check_security_updates() and check_system_updates()

**Comment by fdellwing** [comment]
This should be if at all optional. If we lock an update its because the update will break the site. So it will never be unlocked again until the site is reworked completely. Furthermore it is much more complex because you need to rework your whole code that detects an update and sets the exit code.

**Comment by cytopia** [comment]
OK. I'll merge it. Thanks for the good work!

**Pull request merged**